Source/WebCore/page/LocalFrame.cpp

```diff
@@ -1232,6 +1232,17 @@ DataDetectionResultsStorage& LocalFrame::dataDetectionResults()
 
 #endif
 
+void LocalFrame::frameWasDisconnectedFromOwner() const
+{
+    if (!m_doc)
+        return;
+
+    if (RefPtr window = m_doc->domWindow())
+        window->willDetachDocumentFromFrame();
+
+    protectedDocument()->detachFromFrame();
```

Comment:
Please make sure to wait for debug EWS before landing. It is often not easy to tell if a particular code path can get called by the Document destructor.